src/pubsub/ua_pubsub_writer.c

```diff
@@ -473,6 +487,39 @@ addDataSetField(UA_Server *server, const UA_NodeId publishedDataSet,
     return result;
 }
 
+void
+UA_Server_updateDataSetField(UA_Server *server, const UA_NodeId dsf, UA_Boolean minor, UA_Boolean major) {
+    UA_DataSetField *currentField = UA_DataSetField_findDSFbyId(server, dsf);
+    if(!currentField) {
+        return;
+    }
+
+    if(currentField->configurationFrozen) {
+        UA_LOG_WARNING(&server->config.logger, UA_LOGCATEGORY_SERVER,
+                       "Update DataSetField failed. DataSetField is frozen.");
+        return;
+    }
+
+    UA_PublishedDataSet *pds =
+        UA_PublishedDataSet_findPDSbyId(server, currentField->publishedDataSet);
+    if(!pds) {
+        return;
+    }
+
+    if(pds->configurationFrozen) {
+        UA_LOG_WARNING(&server->config.logger, UA_LOGCATEGORY_SERVER,
+                       "Update DataSetField failed. PublishedDataSet is frozen.");
+        return;
+    }
+
+    /* Update minor version of PublishedDataSet */
+    UA_UInt32 timeDifference = UA_PubSubConfigurationVersionTimeDifference();
+    if(minor)
+        pds->dataSetMetaData.configurationVersion.minorVersion = timeDifference;
+    if(major)
+        pds->dataSetMetaData.configurationVersion.majorVersion = timeDifference;
+}
+
```

Comment:
This code does not match the logic implied by the function name. In addition the user can change the minor/major version directly. I would preffer to have only implicit changes.

Reply:
I removed the flags.